### ISSUE TITLE
ITEMID within IOEVENTS incorrectly linked to D_ITEMS

### ISSUE THREAD

**Issue by vjm261** [MIT-LCP/mimic-code#9]
The minimum ITEMID defined in D_ITEMS to reference the IOEVENTS table is 30001 however, the IOEVENTS table regularly contains events with ITEM_ID values with nchar() equal to 1, 2 or 3 and therefore cannot match with the correct defined item. 

Common examples such as 13, 18, and 56 are either not defined anywhere in D_ITEMS (13 and 18), or wrongly correspond to something from CHARTEVENTS (56 --> Atrial Pacemaker [Sense]) which leads me to believe they actually correspond to 30013 (D5W), 30018 (0.9% Normal Saline), and 30056 (Po Intake) which are correctly described with ORIGINALROUTE as intravenous push, intravenous push and oral respectively within the IOEVENTS table.

I suggest that the documentation is updated as a temporary fix to avoid further confusion and the bug is fixed in the next database build and release.

**Comment by alistairewj** [comment]
Very good spot. Indeed, no ITEMIDs in the IOEVENTS table should have a value < 30000. 

A bit of history: these ITEMIDs originally overlapped and linked to various tables (e.g. ITEMIDs from IOEVENTS linked to D_IOITEMS, ITEMIDs from MEDEVENTS linked to D_MEDITEMS), and they all overlapped. We shifted IO ITEMIDs to 30000-39999, and MED ITEMIDs to 40000-49999. 

Since IOEVENTS does contain ITEMIDs between 30000-39999 and 40000-49999, indicating a successful shift, I am not sure why it also contains ITEMIDs < 500. I will investigate this week. We plan on releasing a new version of the database this week to address these issues, will close this issue once that happens.

**Comment by alistairewj** [comment]
A new version, MIMIC-III v1.2, has been released. This bug is fixed in that version and I recommend you update your data if you haven't already. Thanks for the report!

**Comment by vjm261** [comment]
MIMIC-III v1.2 INPUTEVENTS_CV_DATA_TABLE contains hundreds of rows with ITEMIDs < 500, specifically I have found cases of 51, 63, 172 and 384. 
It appears to be isolated to the CareVue table but unconfirmed

Heres an awk on the raw files for illustration.
```
$ awk -F',' 'OFS="," { if (index($6,"384")==1) {print}}' INPUTEVENTS_MV_DATA_TABLE.csv 
```

**Comment by alistairewj** [comment]
Noooooooooo not again :(

I did a text match of the label for some `ITEMID`s - an attempt to merge them with other, more frequently used ones. The original MEDEVENTS `ITEMID` range overlaps with the CHARTEVENTS `ITEMID` range so we add 30000 - but here I unfortunately forgot to add it!

For now, just add 30000 to any `ITEMID` below 30000 in INPUTEVENTS_CV. This will be fixed next release. And I'm definitely writing an integrity check script this time...

**Comment by alistairewj** [comment]
Just released v1.3. Info here: http://mimic.physionet.org/about/releasenotes/

This time, we have a continuous integration which validates all ITEMIDs are in the correct ranges. Hopefully this bug is gone forever now! Thanks for the diligent reports.
